app/ai/voice/agents/breeze_buddy/managers/calls.py

```
@@ -662,84 +725,96 @@ async def process_backlog_leads():
-                            call_end_time=datetime.now(timezone.utc),
-                        )
-                        await _release_number(
-                            retry_number_to_use.id, retry_number_to_use.provider
+                        if not retry_customer_mobile or not isinstance(
+                            retry_customer_mobile, str
+                        ):
+                            logger.error(
+                                f"Invalid customer_mobile_number for retry lead {locked_lead.id}"
+                            )
+                            await release_lock_on_lead_by_id(locked_lead.id)
+                            continue
```

Ruff (0.15.2)

[warning] `continue` inside `finally` blocks cause exceptions to be silenced

(B012)